pkg/provider/handlers/functions.go

```diff
@@ -231,3 +233,19 @@ func findNamespace(target string, items []string) bool {
 	}
 	return false
 }
+
+func readMemoryLimitFromSpec(spec *specs.Spec) int64 {
```

Comment:
I would like to see a unit test case for each of these scenarios.

I wonder if you can make the code simpler than having 4x `return 0` statements? Such as an if statement with a few clauses?

Reply:
Unit tests added and you are correct it was pretty unreadable, now it is waaay more simple, see the reason above.

Reply:
It looks great, thanks @Shikachuu